File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);
        unordered_map<int,int> hmap;
        for(int i=0;i<nums.size();i++){
            if(hmap.find(target-nums[i])!=hmap.end()) {
                return{i,hmap[target-nums[i]]};
            }
            hmap[nums[i]]=i;
        }
        return{};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> keyind = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int num = nums[i];
            if(keyind.containsKey(target - num)){
                return new int[]{i, keyind.get(target - num)};
            }
            keyind.put(num, i);
        }
        return new int[]{};
    }
}
